bundles/binding/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/MyqData.java

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.myq.internal;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.commons.io.IOUtils;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.JsonProcessingException;
+import org.codehaus.jackson.map.ObjectMapper;
+
+import static org.openhab.io.net.http.HttpUtil.executeUrl;
+
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * This Class handles the Chamberlain myQ http connection.
+ * 
+ * <ul>
+ * <li>userName: myQ Login Username</li>
+ * <li>password: myQ Login Password</li>
+ * <li>sercurityTokin: sercurityTokin for API requests</li>
+ * <li>header: http header data</li>
+ * <li>webSite: url of myQ API</li>
+ * <li>appId: appId for API requests</li>
+ * </ul>
+ * 
+ * @author Scott Hanson
+ * @author Dan Cunningham
+ * @since 1.8.0
+ */
+public class MyqData {
+	static final Logger logger = LoggerFactory.getLogger(MyqData.class);
+
+	private String userName;
+	private String password;
+	private String sercurityTokin;
+
+	private Properties header;
+	private final String webSite = "https://myqexternal.myqdevice.com";
+	private final String appId = "JVM/G9Nwih5BwKgNCjLxiFUQxQijAebyyg8QUHr7JOrP+tuPb8iHfRHKwTmDzHOu";
```

Comment:
Unfortunately we have to use an existing app id that's out in the wild.  This is also what the smart things plugin is doing and I'm assuming the vera plugin as well.  Not ideal, I know.